### ISSUE TITLE
breaking change in tt_to_flextable with respect to column counts

### ISSUE THREAD

**Issue by gmbecker** [insightsengineering/rtables#843]
It appears that the last CRAN release included a breaking change to the behavior of `tt_to_flextable`, where the column counts are appended to the column labels above them, ather than appearing a second line of header as previously (and as they are in other  rtables renderings and internal modeling). 

This breaks code published in the book put out by the RConsortium's R Tables For Regulatory Submission Working Group.

In particular, the code is:

```r
library(rtables)
data("cadcm", package = "random.cdisc.data")
data("cadsl", package = "random.cdisc.data")

one_count_pct_gen <- function(label = NULL) {
    function(x, .N_col) {
       ret <- rcell(length(unique(x)) * c(1, 1/.N_col),
                    format = "xx (xx.x%)")
       if(!is.null(label))
           obj_label(ret) <- label
       ret
    }
}

library(tern)
lyt <- basic_table(show_colcounts = TRUE) |>
  split_cols_by(var = "ARM") |>
  analyze_num_patients(vars = "USUBJID",
                       .stats = "unique",
                       .labels = "At Least One Concomittant Med") |>
  split_rows_by("CMCLAS",
                split_fun = drop_split_levels) |>
  count_occurrences(vars = "CMDECOD")

library(flextable)

tt_to_flextable(build_table(lyt, cadcm, alt_counts_df = cadsl)) |> 
  theme_booktabs() |> 
  font(fontname = "Open Sans") |> 
  bold(i = ~ V2 %in% "", j = 1, bold = TRUE) |> 
  bold(i = 1, j = 1, bold = TRUE) |> 
  align(j = 2:4, align = "center", part = "all") |> 
  set_table_properties(layout = "fixed") |> 
  autofit() |> 
  mk_par(i = 1, j = 1, part = "header",
         as_paragraph(as_chunk("Conmed Example", props = fp_text_default(font.size = 14))))# |> 
  mk_par(i = 2, j = 1, part = "header",
         as_paragraph("Uses the adcm dataset from ", as_b("random.cdisc.data")))
```

which worked in the previous CRAN release and appears in the published version 1.0 of the book. cc @davidgohel relating to Rconsortium/rtrs-wg#144

**Comment by gmbecker** [comment]
In particular we see that the second call to `mk_par` assumes that there are two lines of header material in the `flextable` object that comes out of `tt_to_flextable`, but that is no longer the case.

**Comment by Melkiades** [comment]
This can be done with 
```r
  mk_par(i = 1, j = 1, part = "header",
         as_paragraph(as_chunk("Conmed Example\n", props = fp_text_default(font.size = 14)), 
                      "Uses the adcm dataset from ", as_b("random.cdisc.data")))
```

or top left material

```r
top_left(tbl) <- c(
  "Conmed Example",
  "Uses the adcm dataset from \nrandom.cdisc.data"
)
```
I got specific requests not to have it on a newline, but I think we need both options, so I added a flag for it `counts_position = "new_line"`

**Comment by gmbecker** [comment]
Thanks @Melkiades.

Yeah we took out the formatting for the second line, but the issue was more that it was a breaking change. we worked around it but had to change code in the already published book to do so, which was less than ideal. I'm hoping breaking changes like this can be avoided in the future now that adoption is starting to broaden; there will increasingly be code in the wild that users expect to generally continue to work.
